source/dub/dub.d

```diff
@@ -662,6 +664,21 @@ class Dub {
 			}
 
 			// prepare the list of tested modules
+
+			NativePath mainfile;
+			if (settings.tempBuild)
+				mainfile = getTempFile("dub_test_root", ".d");
+			else {
+				mainfile = m_project.rootPackage.path ~ format(".dub/code/%s-%s-%s-%s-%s_%s_%s_dub_test_root.d", test_config, settings.buildType,
+					settings.platform.platform.join("."),
+					settings.platform.architecture.join("."),
+					settings.platform.compiler, settings.platform.frontendVersion, lbuildsettings.sourceFiles.length);
+			}
```

Comment:
1) I'm aware the code is old, but I would still move this into a separate method. This way it will be easier to add unittests for it.
2) Ideally you also move the bits to generate the package name from settings into a function instead of copying them around. This way it will be a lot easier in the future to find and adjust them
3) I'm aware the code is old, but new additions should try to be better and have a few basic comments for future readers.

Reply:
```d
				mainfile = m_project.rootPackage.path ~ format(".dub/code/%s-%s-%s-%s-%s_%s_%s_dub_test_root.d", test_config, settings.buildType,
					settings.platform.platform.join("."),
					settings.platform.architecture.join("."),
					settings.platform.compiler, settings.platform.frontendVersion, lbuildsettings.sourceFiles.length);
			}
```

Be aware this is not the `.dub/build` folder. In this place I cannot access the `build_id` witch is generated in: https://github.com/dlang/dub/blob/3288bbb65282e1f47f106b195341ef4288c400ed/source/dub/generators/build.d#L330

The `buildsettings` can be changed in a project with dependencies.

Maybe someone has a better idea? I am aware this `.dub/code` solution is not ideal.

Reply:
> In this place I cannot access the build_id witch is generated in: 

What's stopping you from moving this out into a new public function:

```d
string computeBuildName(string config, GeneratorSettings settings)
{
    return format("%s-%s-%s-%s-%s_%s-%s", config, settings.buildType,
			settings.platform.platform.join("."),
			settings.platform.architecture.join("."),
			settings.platform.compiler, settings.platform.frontendVersion, hashstr);
}
```

And calling this from both places?

Furthermore, you would want to hash `lbuildsettings.sourceFiles` - not their length.

Reply:
> Furthermore, you would want to hash lbuildsettings.sourceFiles - not their length.

This is a good idea, thanks.